fennel/sources/sources.py

```diff
@@ -217,20 +217,19 @@ class Kafka(DataSource):
     sasl_mechanism: Optional[str]
     sasl_plain_username: Optional[str]
     sasl_plain_password: Optional[str]
-    sasl_jaas_config: Optional[str]
     verify_cert: Optional[bool]
 
     def _validate(self) -> List[Exception]:
         exceptions: List[Exception] = []
         if self.security_protocol not in [
-            "PLAIN TEXT",
-            "SASL PLAINTEXT",
-            "SASL SSL",
+            "PLAINTEXT",
+            "SASL_PLAINTEXT",
+            "SASL_SSL",
```

Comment:
IIRC this was not being enforced. Can you add a test to confirm?

Reply:
Tested, I think this is a "bug". `_validate` is called in the magic `__post_init__` function, which only works on `@dataclass`, which we don't use. So none of the `_validate` functions was actually called. The fix is relatively big, it's worth another PR.